tests/smoke-npm-grouped.yaml

```diff
@@ -0,0 +1,125 @@
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>7.21.4'
+            - dependency-name: '@babel/preset-typescript'
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>7.21.4'
+            - dependency-name: '@typescript-eslint/eslint-plugin'
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>5.59.1'
+            - dependency-name: '@types/jest'
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.5.1'
+            - dependency-name: '@types/lodash'
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>4.14.194'
+            - dependency-name: '@typescript-eslint/parser'
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>5.59.1'
+            - dependency-name: babel-jest
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.5.0'
+            - dependency-name: babel-plugin-lodash
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>3.3.4'
+            - dependency-name: diff-sequences
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.4.3'
+            - dependency-name: eslint
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>8.39.0'
+            - dependency-name: eslint-plugin-babel
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>5.3.1'
+            - dependency-name: eslint-plugin-import
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>2.27.5'
+            - dependency-name: eslint-plugin-jest
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>27.2.1'
+            - dependency-name: expect
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.5.0'
+            - dependency-name: fetch-factory
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>0.2.1'
+            - dependency-name: jest
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.5.0'
+            - dependency-name: jest-mock
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.5.0'
+            - dependency-name: jest-snapshot
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>29.5.0'
+            - dependency-name: lodash
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>4.17.21'
+            - dependency-name: prettier
+              source: tests/smoke-npm-grouped.yaml
+              version-requirement: '>2.8.8'
+        source:
+            provider: github
+            repo: dependabot/smoke-tests
+            directory: /npm
```

Comment:
```suggestion
            branch: bdragon/smoke-grouped
```

Since this job uses dependencies that are newly added to this branch, we also need to set the branch in the job definition.

With the branch set we see PRs created for each group:

```
+------------------------------------------------------------------------------------------------------------------------------------+
|                                                Changes to Dependabot Pull Requests                                                 |
+---------+--------------------------------------------------------------------------------------------------------------------------+
| created | fetch-factory ( from 0.0.1 to 0.2.1 )                                                                                    |
| created | babel-jest ( from 28.1.1 to 29.5.0 ), @babel/compat-data ( from 7.21.4 to 7.21.7 ), @babel/generator ( from 7.21.4 to... |
| created | babel-jest ( from 28.1.1 to 29.5.0 ), diff-sequences ( from 28.1.1 to 29.4.3 ), eslint-plugin-jest ( from 26.5.3 to 2... |
| created | eslint-plugin-jest ( from 26.5.3 to 27.2.1 )                                                                             |
| created | @types/node ( from 18.16.2 to 18.16.3 )                                                                                  |
| created | caniuse-lite ( from 1.0.30001481 to 1.0.30001482 )                                                                       |
| created | electron-to-chromium ( from 1.4.377 to 1.4.382 )                                                                         |
+---------+--------------------------------------------------------------------------------------------------------------------------+
Dependabot encountered '7' error(s) during execution, please check the logs for more details.
+------------------------------------------------------+
|            Dependencies failed to update             |
+--------------------------------------+---------------+
| babel-plugin-polyfill-corejs3        | unknown_error |
| @typescript-eslint/scope-manager     | unknown_error |
| @typescript-eslint/types             | unknown_error |
| @typescript-eslint/typescript-estree | unknown_error |
| @typescript-eslint/utils             | unknown_error |
| @typescript-eslint/visitor-keys      | unknown_error |
| @typescript-eslint/type-utils        | unknown_error |
+--------------------------------------+---------------+
```

Reply:
I'm not sure why the errors show up as `unknown_error` in the summary. During the job run they are presented as a `Dependabot::NpmAndYarn::FileUpdater::NoChangeError` which I think is because the dependencies are updated to a newer version but because they're transitive, the dependency files we pass in are not modified?

Reply:
I suspect `NoChangeError` is unexpected and gets reported back as unknown to the service as we generally do not expect it.

If the resultant PR is correct, we might need to tolerate this error for grouped updates and step over any loops which produce it 🤔

Reply:
Once this PR is merged into main you can remove the `branch:` key (either as a separate PR or directly committing to main)

Reply:
Thanks for taking a look, @Nishnha!

I added most of these dependencies to the `package.json` on this branch and the job definition has the sha of the head commit on the branch, so I don't think it would be working at all if it were the missing branch. Errors aside, those groups still don't look right—for example, `fetch-factory` should be grouped with `lodash` but it's not.

Reply:
sorry, just seeing your reply now!

I do think the missing branch is the issue - the job definition has `commit: ~` instead of the SHA as far as I can tell. 
At least, the job run worked in the CLI after I included the branch name


As for the groups not looking right I also found it strange! 
The CLI says that `lodash` was updated and I would also expect it to be grouped with `fetch-factory`, but I don't see it in the PR summary... maybe there's another bug at play here?

Here's a gist of my CLI run https://gist.github.com/Nishnha/cd0aab0e146dbe23d78f7b51c9d7daaa